plugins/modules/provision_workflow_manager.py

```diff
@@ -1137,27 +1219,130 @@ def provision_wireless_device(self):
-            self.result['diff'] = self.validated_config
-            self.result['response'] = execution_id
-            self.log(self.result['msg'], "INFO")
-            return self
-        except Exception as e:
-            self.msg = "Error in wireless provisioning: {0}".format(str(e))
-            self.log(self.msg, "ERROR")
-            self.status = "failed"
-            self.result['response'] = self.msg
-            self.check_return_status()
+        device_uid = self.want.get("prov_params")[0].get("device_id")
+        site_name = self.validated_config.get("site_name_hierarchy")
+        primary_ap_location = self.want.get("prov_params")[0].get("primaryManagedAPLocationsSiteIds")
+        secondary_ap_location = self.want.get("prov_params")[0].get("secondaryManagedAPLocationsSiteIds")
+
+        site_exist, site_id = self.get_site_id(site_name)
+        primary_ap_location_site_id_list = []
+        secondary_ap_location_site_id_list = []
+
+        if primary_ap_location:
+            for primary_sites in primary_ap_location:
+                site_exist, primary_ap_location_site_id = self.get_site_id(primary_sites)
+                primary_ap_location_site_id_list.append(primary_ap_location_site_id)
+
+        if secondary_ap_location:
+            for secondary_sites in secondary_ap_location:
+                site_exist, secondary_ap_location_site_id = self.get_site_id(secondary_sites)
+                secondary_ap_location_site_id_list.append(secondary_ap_location_site_id)
+
+        if self.compare_dnac_versions(self.get_ccc_version(), "2.3.5.3") <= 0:
+            try:
+                response = self.dnac_apply['exec'](
+                    family="wireless",
+                    function="provision",
+                    op_modifies=True,
+                    params={"payload": self.want.get("prov_params")}
+                )
+                execution_id = response.get("executionId")
+                self.get_execution_status_wireless(execution_id=execution_id)
+                self.result["changed"] = True
+                self.result['msg'] = "Wireless device provisioned successfully"
+                self.result['diff'] = self.validated_config
+                self.result['response'] = execution_id
+                self.log(self.result['msg'], "INFO")
+                return self
+            except Exception as e:
+                self.msg = "Error in wireless provisioning: {0}".format(str(e))
+                self.log(self.msg, "ERROR")
+                self.status = "failed"
+                self.result['response'] = self.msg
+                self.check_return_status()
+        else:
+
+            is_device_assigned = self.is_device_assigned_to_site(device_uid)
+            if not is_device_assigned:
+                self.assign_device_to_site([device_uid], site_name, site_id)
+            if secondary_ap_location or secondary_ap_location:
+                try:
```

Comment:
self.log("Assigning managed AP locations for the WLC", "INFO")

Reply:
updated the code